server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -315,10 +316,22 @@ public boolean offer(E e) {
      * A handler for rejected tasks that adds the specified element to this queue,
      * waiting if necessary for space to become available.
      */
-    static class ForceQueuePolicy implements XRejectedExecutionHandler {
+    static class ForceQueuePolicy extends EsRejectedExecutionHandler {
+
+        private final boolean rejectAfterShutdown;
```

Comment:
I wonder if the end goal here is to remove this boolean completely to never accept a request after shutdown? Seems dangerous since it may just be ignored.

I am fine with the approach of incrementally trying it out (also to validate the the original test failure is gone), but if this is an interim step, it would be good to add a comment here.

Reply:
I agree it would be great to remove the `rejectAfterShutdown` flag and have all thread pool handling rejections even after shutdown or termination, but this is quite some work specially for `generic`. 

I added a comment in cb3b486ebebeb0e74ccf0f3a989f9633ace8ac3a, let me know if that's enough. We can also add a specific assertion (and deprecate some scaling builder) so that we only allow a fixed list of thread pools to have this flag set to `false` in order to avoid introducing more thread pools that do not handle rejections in all cases.